v1/src/main/java/com/google/cloud/teleport/templates/common/SpannerConverters.java

```diff
@@ -606,4 +817,46 @@ static TimestampBound getTimestampBound(String timestamp) {
       return TimestampBound.ofReadTimestamp(tsVal);
     }
   }
+
+  /** Interface to validate Struct. If validation fails then throws RuntimeException */
+  interface StructValidator {
+    void validate(Struct struct);
+  }
+
+  /**
+   * To validate struct and check whether data can be exported in the format supported by Vertex AI
+   * Vector Search Index.
+   */
+  public static class VectorSearchStructValidator implements StructValidator {
+    private static final String ID = "id";
+    private static final String EMBEDDING = "embedding";
+
+    @Override
+    public void validate(Struct struct) {
+      // ID type Check
+      // ToDo: Add validation for `restricts` & `crowding`.
+      validateType(struct, EMBEDDING, Type.array(Type.float64()));
+
+      // Not Null
+      validateNotNull(struct, ID);
+      validateNotNull(struct, EMBEDDING);
```

Comment:
We are expecting these 2 columns to be always present. We should add that these 2 columns are compulsary in the description of columnsToExport parameter.